Formula/m/mummer.rb

```diff
@@ -0,0 +1,44 @@
+# Formula for installing the mummer aligner. It must be compiled with GCC from
+# Homebrew, and not clang or gcc from macOS (which is really clang), to obtain a
+# valid executable. Clang+libstdc++ seem to have issues: it compiles, but the
+# executable obtained is buggy (does not pass tests).
+class Mummer < Formula
+  desc "Genome alignment tool"
+  homepage "https://github.com/mummer4/mummer"
+  url "https://github.com/mummer4/mummer/releases/download/v4.0.1/mummer-4.0.1.tar.gz"
+  sha256 "bc20ae2701a0b2e323e4e515b7cfa18a0f0cb34a4ff5844b289b2de0154e3d3e"
+  license "Artistic-2.0"
+
+  depends_on "gcc" => [:build, :test]
+
+  def install
+    system "./configure", "--disable-silent-rules", *std_configure_args, "CC=gcc-14", "CXX=g++-14"
```

Comment:
```suggestion
    ENV["CC"] = "gcc-14"
    ENV["CXX"] = "g++-14"
    system "./configure", "--disable-silent-rules", *std_configure_args
```

Reply:
Should be using compiler selection[^1] instead and comment should be moved to `cause`, e.g.
```rb
fails_with :clang do
  cause <<~EOS
    Clang+libstdc++ seem to have issues: ...
  EOS
end
```

Linux already uses GCC, so usually shouldn't need brew GCC unless there is a Fortran dependency.

[^1]: https://docs.brew.sh/Formula-Cookbook#compiler-selection

Reply:
I committed this suggestion. But it cause against the recommendation of [autoconf](https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Defining-Variables.html#Defining-Variables)

Reply:
Thanks for the suggestion regarding `fails_with`. That's how it is implemented now.